libraries/SITL/SIM_Vicon.cpp

```diff
@@ -204,6 +204,34 @@ void Vicon::update_vicon_position_estimate(const Location &loc,
             NULL, 0);
         msg_buf[msg_buf_index].time_send_us = time_send_us;
     }
+
+    static uint64_t prev_time_us = now_us + time_offset_us;
+    static Vector3f prev_position(pos_corrected.x, pos_corrected.y, pos_corrected.z);
+    static Vector3f prev_angle(roll, pitch, yaw);
+
+    float postion_delta[3] = {prev_position.x - pos_corrected.x, 
+                                prev_position.y - pos_corrected.y,
+                                prev_position.z - pos_corrected.z};
+    float angle_delta[3] = {prev_angle.x - roll, prev_angle.y - pitch, prev_angle.z - yaw};
```

Comment:
I think that the angular delta is probably not as simple as this because this would lead to large jumps when the vehicle crosses through the +180 <-> -180 point for roll and yaw.  I think we need some advice from people like @chobitsfan, @tridge or @priseborough about how to get this angle.

My first guess is that we would need to convert the attitudes to quaternions and then use the "angular_difference" method and then convert the result to euler angles.

Reply:
You are totally right. What I had would lead to some not nice jumps!

I implemented your suggestions. Though I do wonder if there is a simpler method as we are talking about rather small angle changes? Do you know which message of vicon or T265 relates closest to the delta message? I couldn't find much in the SDK docs.

Reply:
@hendjoshsr71 here's Thien's script which pulls data from the T265 and sends it to AP using the position-delta message: https://github.com/thien94/vision_to_mavros/blob/master/scripts/t265_to_mavlink.py#L299.  It's not immediately obvious to me if this is helpful.

I suspect that if it flies OK in the simulator then what you've got is probably close to correct.

I'll ask Paul Riseborough if he can spare a couple of minutes to peek at what you've done.

Reply:
Thanks! This link was quite helpful to see the transformation matrix being used as the definitions from what frame things were being referenced and origins being used for the computations.